src/player.cpp

```diff
@@ -11013,6 +11013,8 @@ void player::forced_dismount()
     bool mech = false;
     if( mounted_creature ) {
         auto mon = mounted_creature.get();
+        // monster has been storing up moves while riding, reset them now so it dosnt zoom off.
```

Comment:
Would be a much better fix if the monster would *not* store up their moves. The current fix just hides the problem after it has already been created.

Reply:
Yep, this was my original fix, Other devs advised that it might be cleaner to put it all here.
But I can just add a check for mountedness in creature.cpp where moves are regenerated.